tests/unit/utils/retrieveDefaultRead.js

```diff
@@ -0,0 +1,70 @@
+const assert = require('assert');
+
+const { ConfigObject } = require('../../../lib/Config');
+const retrieveDefaultRead =
+    require('../../../lib/api/apiUtils/object/retrieveDefaultRead');
+
+const locationConstraints = {  // eslint-disable-line quote-props
+    bucketmatch: {
+        type: 'aws_s3',
+        legacyAwsBehavior: true,
+        details: {
+            bucketMatch: true,
+        },
+    },
+    nobucketmatch: {
+        type: 'aws_s3',
+        legacyAwsBehavior: true,
+        details: {
+            bucketMatch: false,
+        },
+    },
+    'us-east-1': {
+        type: 'file',
+        legacyAwsBehavior: true,
+        details: {},
+    },
+};
+
+const bucket = 'testBucket';
+const key = 'objectKey';
+
+describe('Testing retrieveDefaultRead', () => {
+    let config;
+
+    before(() => {
+        config = new ConfigObject();
+        config.setLocationConstraints(locationConstraints);
+    });
+
+    const tests = [
+        {
+            it: 'should return null location constraints does not exist',
+            loc: 'nonexistloc',
+            results: null,
+        },
+        {
+            it: 'should return correct results for bucketMatch true location',
+            loc: 'bucketmatch',
+            results: {
+                location: 'bucketmatch',
+                key,
+                locationType: 'aws_s3',
+            },
+        },
+        {
+            it: 'should return correct results for bucketMatch false location',
+            loc: 'nobucketmatch',
+            results: {
+                location: 'nobucketmatch',
+                key: `${bucket}/${key}`,
+                locationType: 'aws_s3',
+            },
+        },
+    ];
+
+    tests.forEach(test => it(test.it, () => {
```

Comment:
I usually like `forEach` loops for test factorization, here it seems duplication is minimal and does not really justify the use of a loop compared to explicit separate tests (for readability and it's then easier to add custom checks to each if needed).